### PR TITLE
Update cookie cutter post nyc tutorial

### PR DESCRIPTION
**Pull request by mrocklin**
This includes a few changes:

1.  Use nomkl in the environment.yml file
2.  Apt-get graphviz
3.  Upgrade JupyterLab 0.35, and also hub-labextension to match
4.  Add a default workspace that brings up the dashboard plots automatically
5.  Add a Welcome.md file that is displayed in the JLab environment

cc @jhamman @rsignell-usgs 

A similar setup is available [here](http://binder.pangeo.io/v2/gh/mrocklin/pydata-nyc-2018-tutorial/master)

### CODE REVIEW

**Review by jhamman — APPROVED** [review]
LGTM. Thanks @mrocklin.

**Comment by jhamman** [comment]
> Upgrade JupyterLab 0.35, and also hub-labextension to match

I don't see the hub-labextension. I'm also not sure this works with binder currently.

File: {{cookiecutter.project_slug}}/binder/postBuild [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 jupyter serverextension enable --py nbserverproxy --sys-prefix
 jupyter labextension install @jupyter-widgets/jupyterlab-manager \
-                             @jupyterlab/hub-extension \
+                             @jupyterlab/hub-extension@0.12 \
```
Comment (mrocklin) [inline {{cookiecutter.project_slug}}/binder/postBuild:4]: cc @jhamman

Comment (jhamman) [reply {{cookiecutter.project_slug}}/binder/postBuild:4]: Thanks. I've been having weird browser issues with github today. I blame it on that 😉!

Comment (mrocklin) [reply {{cookiecutter.project_slug}}/binder/postBuild:4]: FYI, github is experiencing a service outage.  It's not you.  It's everyone.

**Comment by mrocklin** [comment]
I would like to try this before merging.  I tried to install it locally but found that there wasn't a setup.py file (#8)

**Comment by mrocklin** [comment]
Works well.  Merging this in the morning.

**Comment by rsignell-usgs** [comment]
Should we also specify `jupyterlab-manager@0.38` version to ensure [compatibility](https://github.com/jupyter-widgets/ipywidgets/tree/master/packages/jupyterlab-manager#version-compatibility)?    
(I'm also surprised the @0.35/@0.38 combo works.  I couldn't get this to work last week).

**Comment by mrocklin** [comment]
What's here works well today.  I'm inclined to merge this known good state.  

I agree that we should probably start pinning versions more explicitly.  I'm curious, does `jupyter labextension install` take into account dependency constraints between JLab and the extensions?  If so then it might make sense to *not* pin versions here.  I'm not sure.

@rsignell-usgs have you tried `jupyterlab-manager@0.38` with the current setup in this PR?  Are you able to verify that it works?

**Comment by rsignell-usgs** [comment]
@mrocklin , actually, when I tried to pin `jupyterlab-manager@0.38` with `jupyterlab=0.35` it *didn't work*, which was confusing to me since I thought that's what this chart was telling me to do: https://github.com/jupyter-widgets/ipywidgets/tree/master/packages/jupyterlab-manager#version-compatibility

**Comment by rabernat** [comment]
I just tried this branch to create my first pangeo binder: https://github.com/rabernat/pangeo_esgf_demo

I created it using the command 
```
cookiecutter https://github.com/pangeo-data/cookiecutter-pangeo-binder.git --checkout post-nyc-tutorial
```

I only made one tiny change to the environment: https://github.com/rabernat/pangeo_esgf_demo/blob/master/binder/environment.yml#L30
(installing xarray from pip / git)

I am not able to create a KubeCluster
```python
from dask_kubernetes import KubeCluster
cluster = KubeCluster(n_workers=2)
```

```
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
<ipython-input-1-d43e26c2e614> in <module>()
      1 # but this does not
      2 from dask_kubernetes import KubeCluster
----> 3 cluster = KubeCluster(n_workers=2)
      4 kube_client = Client(cluster)
      5 display(cluster)

/srv/conda/lib/python3.6/site-packages/dask_kubernetes/core.py in __init__(self, pod_template, name, namespace, n_workers, host, port, env, **kwargs)
    164             fn = dask.config.get('kubernetes.worker-template-path')
    165             fn = fn.format(**os.environ)
--> 166             with open(fn) as f:
    167                 d = yaml.safe_load(f)
    168             pod_template = make_pod_from_dict(d)

FileNotFoundError: [Errno 2] No such file or directory: '/home/jovyan/.dask/config.yaml'
```

Could this be because I created a local cluster first? See notebook:
https://github.com/rabernat/pangeo_esgf_demo/blob/master/notebooks/test_opendap_with_distributed.ipynb

**Comment by jhamman** [comment]
@rabernat - I think your repo is missing the `.dask` directory. Perhaps you missed that in your `git add`?

**Comment by rabernat** [comment]
Yes ok that makes sense. 

But is there any reason not to have that as part the cookie cutter? I expected the cookie cutter to take care of all dask stuff for me.

**Comment by jhamman** [comment]
(sorry @mrocklin for taking your PR off topic. I think we can merge this in its current state)

@rabernat - I don't follow. It is part of the cookie cutter. I think you must have created a git repo without it:

```bash
 ~/workdir  cookiecutter https://github.com/pangeo-data/cookiecutter-pangeo-binder.git --checkout post-nyc-tutorial
You've downloaded /Users/jhamman/.cookiecutters/cookiecutter-pangeo-binder before. Is it okay to delete and re-download it? [yes]: yes
full_name [Joe Hamman]:
email [joe@example.com]:
github_username [jhamman]:
project_name [Pangeo-Binder-Example]: junk222
project_slug [junk222]:
project_short_description [junk222 contains all the boilerplate you need to create a Pangeo-Binder ready repository.]:
open_source_license [BSD license]:
 ~/workdir  cd junk222
 ✘  ~/workdir/junk222  ll -a
total 32
drwxr-xr-x  91 jhamman  staff   2.8K Oct 23 20:10 ..
-rw-r--r--   1 jhamman  staff   1.5K Oct 23 20:10 LICENSE
-rw-r--r--   1 jhamman  staff   907B Oct 23 20:10 Welcome.md
-rw-r--r--   1 jhamman  staff   1.1K Oct 23 20:10 .gitignore
drwxr-xr-x   9 jhamman  staff   288B Oct 23 20:10 .
-rw-r--r--   1 jhamman  staff   453B Oct 23 20:10 README.rst
drwxr-xr-x   7 jhamman  staff   224B Oct 23 20:10 binder
drwxr-xr-x   3 jhamman  staff    96B Oct 23 20:10 .dask
drwxr-xr-x   3 jhamman  staff    96B Oct 23 20:10 notebooks
 ~/workdir/junk222  git st
fatal: not a git repository (or any of the parent directories): .git
```

I'm not sure if we can make cookiecutter make a git repo for you but if we can, that would be a cool feature.

By the way, the dask directory in the cookiecutter is here: https://github.com/pangeo-data/cookiecutter-pangeo-binder/blob/master/%7B%7Bcookiecutter.project_slug%7D%7D/.dask/config.yaml

After adding this to your notebook, your `KubeCluster` cell runs to completion in about 4 seconds.

**Comment by rabernat** [comment]
You're absolutely right. I forgot to add the hidden `.dask` directory to my repo.

**Comment by mrocklin** [comment]
I'm curious, did `git add .` not include the hidden directory by default?

Presumably others might run into your same situation.  Are there ways that
we can make this friendlier?

**Comment by rabernat** [comment]
I didn't run that command. I manually added the directories. A poor choice
on my part.

**Pull request merged**